src/snowflake/connector/storage_client.py

```diff
@@ -185,7 +185,7 @@ def get_file_header(self, filename: str) -> FileHeader | None:
     def preprocess(self) -> None:
         meta = self.meta
         logger.debug(f"Preprocessing {meta.src_file_name}")
-
+        self.get_digest()
         if not meta.overwrite:
             self.get_file_header(meta.dst_file_name)  # Check if file exists on remote
```

Comment:
while this `self.get_file_header` raise error? if it raises error, then is it still correct to call `self.get_digest` ahead

Reply:
As discussed offline, update the PR to add a new call to `get_digest` under `if not meta.overwrite:` and keep the previous `get_digest`.